### PR TITLE
unpack_input decorator for tf_convnext

### PR DESCRIPTION
**Pull request by johko**
# What does this PR do?

Add the unpack_input decorator for convnext (see #16051) 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@Rocketknight1

### CODE REVIEW

File: src/transformers/models/convnext/modeling_tf_convnext.py [hunk 1]
```diff
@@ -300,29 +301,16 @@ def call(
...
-            input_ids=pixel_values,
-            output_hidden_states=output_hidden_states,
-            return_dict=return_dict,
-            training=training,
-            kwargs_call=kwargs,
-        )
-
-        if "input_ids" in inputs:
-            inputs["pixel_values"] = inputs.pop("input_ids")
-
```
Comment (johko) [inline src/transformers/models/convnext/modeling_tf_convnext.py:315]: Was a bit unsure about the removal of the upper two lines (313,314), but according to the tests it seems safe to delete them, as they don't seem to serve anymore purpose now

Comment (gante) [reply src/transformers/models/convnext/modeling_tf_convnext.py:315]: Correct, with the decorator these lines are not needed 😎

File: src/transformers/models/convnext/modeling_tf_convnext.py [hunk 2]
```diff
@@ -445,6 +433,7 @@ def __init__(self, config, *inputs, add_pooling_layer=True, **kwargs):
 
     @add_start_docstrings_to_model_forward(CONVNEXT_INPUTS_DOCSTRING)
     @replace_return_docstrings(output_type=TFBaseModelOutputWithPooling, config_class=_CONFIG_FOR_DOC)
+    @unpack_inputs
```
Comment (gante) [inline src/transformers/models/convnext/modeling_tf_convnext.py:436]: Sorry to be a pain about this -- in the other PRs, the new decorator is the top decorator. For consistency, can you please move it so as to be the top decorator? 🤗  (there is another instance of this in these changes)

**Review by gante — APPROVED** [review]
Fantastic, thank you so much for the contribution 🔥  Have a look at the comment below, and let me know if the tests passed (`RUN_SLOW=1 py.test -vv tests/[model_name]/test_modeling_tf_[model_name].py`). 

Will merge as soon as these 2 comments are addressed!

**Comment by johko** [comment]
Sure, no problem @gante . The tests ran without a problem.

**Pull request merged**